### ISSUE TITLE
[Documentation Update]: SFR3 - Category: Telemetry - Deployment/Usage Telemetry

### ISSUE THREAD

**Issue by jbinko** [Azure/Azure-Verified-Modules#1618]
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Description

The section on telemetry ID can be quite overwhelming for new contributors.
For clarity, contributors do not need to create their own telemetry ID.
Instead, they should use the pre-assigned telemetry ID, which can be found in the BicepResourceModules.csv file on GitHub at this link: https://github.com/Azure/Azure-Verified-Modules/blob/main/docs/static/module-indexes/BicepResourceModules.csv.

The main task is to incorporate this assigned telemetry ID correctly within a bicep file.
Simplifying this information can help contributors understand the process more easily.

When I first read it, I attempted to create my own telemetry ID for the module based on the description, only to later realize that I needed to use the one assigned in the CSV file.

I also recommend adding a Bicep template example for contributors. It's important to include the `take` function to restrict the telemetry ID to a maximum of 64 characters, as this is already being implemented in some modules and seems practical.

```Bicep
#disable-next-line no-deployments-resources
resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
  name: take(
    '46d3xbcp.res.elasticsan-elasticsan.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}',
    64
  )
  properties: {
    mode: 'Incremental'
    template: {
      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
      contentVersion: '1.0.0.0'
      resources: []
      outputs: {
        telemetry: {
          type: 'String'
          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
        }
      }
    }
  }
}
```

**Comment by jbinko** [comment]
Maybe we can in the section https://azure.github.io/Azure-Verified-Modules/specs/shared/#bicep
point them to: https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpfr4---category-composition---telemetry-enablement
so they can early understand they do not have to re-invent the code and that telemetry ID will be assigned.

**Comment by matebarabas** [comment]
Hi, @jbinko, good suggestion on better explaining that module owners should not create their own telemetry IDs but take the one provided in the CSV. I can help with this part.

I'd leave it on the @Azure/avm-core-team-technical-bicep however to decide whether we should adopt the snippet you referenced. Especially because the segment defined in the CSV has a length limit already applied that should guarantee that we still fit the 64 char overall limit, even after adding the suffixes.

**Comment by AlexanderSehr** [comment]
> Hi, [@jbinko](https://github.com/jbinko), good suggestion on better explaining that module owners should not create their own telemetry IDs but take the one provided in the CSV. I can help with this part.
> 
> I'd leave it on the [@Azure/avm-core-team-technical-bicep](https://github.com/orgs/Azure/teams/avm-core-team-technical-bicep) however to decide whether we should adopt the snippet you referenced. Especially because the segment defined in the CSV has a length limit already applied that should guarantee that we still fit the 64 char overall limit, even after adding the suffixes.

I'm a bit torn. Just checked and 25 modules implement `take()`, 277 don't. 
In general, using the function makes sense to definitely keep within the contraints of a deployment name. However, I can't help but also like the idea of the deployment failing **if** the person that created the telemetry ID messed up and did not stay within the limit of its unique segment. If there's not `take()` function it would at least throw an error and this could be fixed. 
Both options have their merits.

@eriqua / @ReneHezser / @ChrisSidebotham / @jtracey93 / @rahalan any thoughts?

**Comment by matebarabas** [comment]
> Hi, [@jbinko](https://github.com/jbinko), good suggestion on better explaining that module owners should not create their own telemetry IDs but take the one provided in the CSV. I can help with this part.

I've addressed this first part in this PR: https://github.com/Azure/Azure-Verified-Modules/pull/2405

**Comment by matebarabas** [comment]
> > Hi, [@jbinko](https://github.com/jbinko), good suggestion on better explaining that module owners should not create their own telemetry IDs but take the one provided in the CSV. I can help with this part.
> > I'd leave it on the [@Azure/avm-core-team-technical-bicep](https://github.com/orgs/Azure/teams/avm-core-team-technical-bicep) however to decide whether we should adopt the snippet you referenced. Especially because the segment defined in the CSV has a length limit already applied that should guarantee that we still fit the 64 char overall limit, even after adding the suffixes.
> 
> I'm a bit torn. Just checked an 25 modules implement `take()`, 277 don't. In general, using the function makes sense to definitely keep within the contraints of a deployment name. However, I can't help but also like the idea of the deployment failing **if** the person that created the telemetry ID messed up and did not stay within the limit of its unique segment. If there's not `take()` function it would at least throw an error and this could be fixed. Both options have their merits.
> 
> [@eriqua](https://github.com/eriqua) / [@ReneHezser](https://github.com/ReneHezser) / [@ChrisSidebotham](https://github.com/ChrisSidebotham) / [@jtracey93](https://github.com/jtracey93) / [@rahalan](https://github.com/rahalan) any thoughts?

I agree to the approach of letting the problem surface if the combined ID (with the suffix added) is too long.

**Comment by eriqua** [comment]
> > Hi, [@jbinko](https://github.com/jbinko), good suggestion on better explaining that module owners should not create their own telemetry IDs but take the one provided in the CSV. I can help with this part.
> > I'd leave it on the [@Azure/avm-core-team-technical-bicep](https://github.com/orgs/Azure/teams/avm-core-team-technical-bicep) however to decide whether we should adopt the snippet you referenced. Especially because the segment defined in the CSV has a length limit already applied that should guarantee that we still fit the 64 char overall limit, even after adding the suffixes.
> 
> I'm a bit torn. Just checked and 25 modules implement `take()`, 277 don't. In general, using the function makes sense to definitely keep within the contraints of a deployment name. However, I can't help but also like the idea of the deployment failing **if** the person that created the telemetry ID messed up and did not stay within the limit of its unique segment. If there's not `take()` function it would at least throw an error and this could be fixed. Both options have their merits.
> 
> [@eriqua](https://github.com/eriqua) / [@ReneHezser](https://github.com/ReneHezser) / [@ChrisSidebotham](https://github.com/ChrisSidebotham) / [@jtracey93](https://github.com/jtracey93) / [@rahalan](https://github.com/rahalan) any thoughts?

I’d also suggest removing the `take` function from the telemetry snippet and letting the truncation happen in the CSV instead. Using take might cut off the unique ID suffix, which could make the deployment name non-unique. In the worst case, it could even truncate the telemetry ID itself, making the telemetry hard to recognize.

**Comment by AlexanderSehr** [comment]
If we end up agreeing on this I'd suggest to update the 25 modules that use it today and add a static test that validates the syntax (and maybe even the length)
